docs/contributing/smart-component.md

```diff
@@ -2,9 +2,191 @@
 kind: '👋 Contributing'
 title: 'Smart Component guidelines'
 ---
-
 # Smart Component guidelines
 
-<cc-notice intent="warning" message="This doc is a work in progress..."></cc-notice>
+Smart components connect UI components with data-fetching logic and API calls. This guide will help you create smart components that handle state management and communicate with backend services.
+
+## Building a Smart Component
+
+### 1. Selector and Context Parameters
+
+First, define your smart component with a selector and context parameters:
+
+```js
+defineSmartComponent({
+  selector: 'my-component', // CSS selector used to find and bind to all instances of this component in the DOM
+  params: {
+    // Define expected context parameters
+    apiConfig: { type: Object },
+    userId: { type: String },
+    isAdmin: { type: Boolean, optional: true }
+  },
+  onContextUpdate: ...
+});
+```
+A smart component reacts to changes in its context by receiving updates through the onContextUpdate callback.
+The params section defines the specific pieces of context data that the component needs to function and interact with APIs.
+Parameters can be marked as optional with `optional: true` if they are not strictly required.
+At least one parameter must be mandatory (non-optional) - you cannot make all parameters optional.
+The `type` values inside `params` serve as documentation and have no functional impact on the component's behavior.
+
+### 2. State Updates and Data Fetching
+
+The `updateComponent` helper is a powerful tool for managing component state:
+
+```js
+// Pass a direct value to set the property
+updateComponent('state', { state: 'loading' });
+
+// Pass a function to modify the property with Immer
+updateComponent('state', (state) => {
+  state.saving = true;
+  state.lastUpdate = Date.now();
+});
+```
+
+Using Immer under the hood, `updateComponent` lets you write immutable updates in a mutable style.
```

Comment:
@hsablonniere with the problems we faced with the creation tunnel, it might be a good idea to talk about the auto freeze? :thinking:

Reply:
Ok I'll wait for @hsablonniere 's input before modifying and merging since I don't know enough about the issue :+1: